File: cpp/1196.cpp
#include <iostream>
#include <queue>
#include <string>
#include <unordered_map>

using namespace std;

int main()
{
    string keyboard = "`1234567890-=QWERTYUIOP[]\\ASDFGHJKL;'ZXCVBNM,./";
    unordered_map<char, char> keys;
    for (int i = 1; i < keyboard.length(); i++) {
        keys[keyboard[i]] = keyboard[i - 1];
    }
    keys['\n'] = '\n';
    keys[' '] = ' ';
    queue<char> text;
    string line;
    while (getline(cin, line)) {
        for (char i: line) {
            text.push(keys[i]);
        }
        text.push('\n');
    }
    while (!text.empty()) {
        cout << text.front();
        text.pop();
    }
    return 0;
}

File: py/1196.py
from collections import deque
from sys import stdout


def main() -> None:
    print = stdout.write
    keyboard = "`1234567890-=QWERTYUIOP[]\\ASDFGHJKL;'ZXCVBNM,./"
    keys = dict()
    for i in range(1, len(keyboard)):
        keys[keyboard[i]] = keyboard[i - 1]
    keys['\n'] = '\n'
    keys[' '] = ' '
    text = deque()
    while True:
        try:
            text.extend([i for i in input()])
            text.append('\n')
        except EOFError:
            break
    while text:
        print(keys[text.popleft()])



if __name__ == '__main__':
    main()
